packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -376,7 +377,7 @@ class IOSDevice extends Device {
           packageId,
           this,
           ipv6,
-          debuggingOptions.observatoryPort,
+          debuggingOptions.hostVmservicePort,
```

Comment:
The device port could also be used to filter results from an mdns query. It looks like currently `MDnsObservatoryDiscovery` will throw a tool exit if there is more than one result, asking for an appId so that it can refine the mdns query, but another way to distinguish would be to pass the device port to the engine and then filter on it. Maybe add a TODO.